near.js

```diff
@@ -97,18 +97,37 @@ class Near {
             functionCall.amount = amount;
         }
 
-        const buffer = FunctionCallTransaction.encode(functionCall).finish();
+        return this.submitTransaction(originator, functionCall);
+    }
+
+    async submitTransaction(originator, transaction) {
+        const protoClass = transaction.constructor;
+        const className = protoClass.name;
+        const propertyName = className[0].toLowerCase() + className.replace(/Transaction$/, '').substring(1);
+        console.log('submitTransaction', className, propertyName, originator);
+
+        const buffer = protoClass.encode(transaction).finish();
         const signatureAndPublicKey = await this.nearClient.signer.signBuffer(
             buffer,
             originator,
         );
 
         const signedTransaction = SignedTransaction.create({
-            functionCall,
+            [propertyName]: transaction,
```

Comment:
What is this syntax `[propertyName]` as a field name?

Reply:
it uses value of `propertyName` as a field name